diff --git a/server/app.ts b/server/app.ts
index eddaa67..7b8ed99 100644
--- a/server/app.ts
+++ b/server/app.ts
@@ -5,11 +5,9 @@ import createError from 'http-errors'
 import nunjucksSetup from './utils/nunjucksSetup'
 import errorHandler from './errorHandler'
 import { appInsightsMiddleware } from './utils/azureAppInsights'
-import authorisationMiddleware from './middleware/authorisationMiddleware'
 
-import setUpAuthentication from './middleware/setUpAuthentication'
+import setUpPrisonerAuth from './middleware/setUpPrisonerAuth'
 import setUpCsrf from './middleware/setUpCsrf'
-import setUpCurrentUser from './middleware/setUpCurrentUser'
 import setUpHealthChecks from './middleware/setUpHealthChecks'
 import setUpStaticResources from './middleware/setUpStaticResources'
 import setUpWebRequestParsing from './middleware/setupRequestParsing'
@@ -33,10 +31,8 @@ export default function createApp(services: Services): express.Application {
   app.use(setUpWebRequestParsing())
   app.use(setUpStaticResources())
   nunjucksSetup(app)
-  app.use(setUpAuthentication())
-  app.use(authorisationMiddleware())
+  app.use(setUpPrisonerAuth())
   app.use(setUpCsrf())
-  app.use(setUpCurrentUser())
 
   app.use(routes(services))
 
diff --git a/server/interfaces/hmppsUser.ts b/server/interfaces/hmppsUser.ts
index f0a6dd2..8c01550 100644
--- a/server/interfaces/hmppsUser.ts
+++ b/server/interfaces/hmppsUser.ts
@@ -1,4 +1,6 @@
-export type AuthSource = 'nomis' | 'delius' | 'external' | 'azuread'
+import { LaunchpadUser } from '@ministryofjustice/hmpps-prisoner-auth'
+
+export type AuthSource = 'nomis' | 'delius' | 'external' | 'azuread' | 'prisoner-auth'
 
 /**
  * These are the details that all user types share.
@@ -57,4 +59,4 @@ export interface AzureADUser extends BaseUser {
   authSource: 'azuread'
 }
 
-export type HmppsUser = PrisonUser | ProbationUser | ExternalUser | AzureADUser
+export type HmppsUser = PrisonUser | ProbationUser | ExternalUser | AzureADUser | LaunchpadUser
diff --git a/server/middleware/setUpPrisonerAuth.ts b/server/middleware/setUpPrisonerAuth.ts
new file mode 100644
index 0000000..e1df309
--- /dev/null
+++ b/server/middleware/setUpPrisonerAuth.ts
@@ -0,0 +1,77 @@
+import passport from 'passport'
+import flash from 'connect-flash'
+import { Router } from 'express'
+import { LaunchpadUser, PrisonerAuth, minutes } from '@ministryofjustice/hmpps-prisoner-auth'
+import { HmppsUser } from '../interfaces/hmppsUser'
+
+passport.serializeUser((user, done) => {
+  // Not used but required for Passport
+  done(null, user)
+})
+
+passport.deserializeUser((user, done) => {
+  // Not used but required for Passport
+  done(null, user as Express.User)
+})
+
+const prisonerAuth = new PrisonerAuth({
+  launchpadAuthUrl: 'VALUE_HERE',
+  clientId: 'VALUE_HERE',
+  clientSecret: 'VALUE_HERE',
+  tokenMinimumLifespan: minutes(5),
+})
+
+passport.use('prisoner-auth', prisonerAuth.passportStrategy())
+
+export default function setUpPrisonerAuth() {
+  const router = Router()
+
+  router.use(passport.initialize())
+  router.use(passport.session())
+  router.use(flash())
+
+  router.get('/autherror', (req, res) => {
+    res.status(401)
+    return res.render('autherror')
+  })
+
+  router.get('/sign-in', passport.authenticate('prisoner-auth'))
+
+  router.get('/sign-in/callback', (req, res, next) =>
+    passport.authenticate('prisoner-auth', {
+      successReturnToOrRedirect: req.session.returnTo || '/',
+      failureRedirect: '/autherror',
+    })(req, res, next),
+  )
+
+  router.use('/sign-out', (req, res, next) => {
+    if (req.user) {
+      req.logout(err => {
+        if (err) return next(err)
+        return req.session.destroy(() => res.redirect('/'))
+      })
+    } else res.redirect('/')
+  })
+
+  router.use(async (req, res, next) => {
+    if (!req.isAuthenticated()) {
+      req.session.returnTo = req.originalUrl
+      return res.redirect('/sign-in')
+    }
+
+    return prisonerAuth
+      .validateAndRefreshUser(req.user as LaunchpadUser)
+      .then(user => {
+        req.user = user
+        next()
+      })
+      .catch(() => res.redirect('/autherror'))
+  })
+
+  router.use((req, res, next) => {
+    res.locals.user = req.user as HmppsUser
+    next()
+  })
+
+  return router
+}
